cmd/env.go

```diff
@@ -361,3 +337,38 @@ func commonEnvFlags(flags *pflag.FlagSet) (server, namespace, context string, er
 
 	return server, namespace, context, nil
 }
+
+func resolveEnvFlags(flags *pflag.FlagSet) (server, namespace string, err error) {
+	server, ns, ctx, err := commonEnvFlags(flags)
+	if err != nil {
+		return "", "", err
+	}
+
+	if len(server) == 0 {
+		// If server is not provided, use the provided context.
+		// If context is also not provided, use the current context.
+		var context *string
+		if len(ctx) != 0 {
+			context = &ctx
+		}
+
+		server, namespace, err = resolveContext(context)
+		if err != nil {
+			return "", "", err
+		}
+
+		if flags.Changed(flagEnvNamespace) {
+			// if a namespace is specified, we don't want to use the default one
+			namespace = ns
+		}
+	} else {
+		if len(ns) == 0 {
+			namespace = defaultNamespace
```

Comment:
By my count we're setting `namespace = defaultNamespace` in 3 different places. It's kind of a maze right now (and it's my own fault).

I feel like we're going to shoot ourselves in the foot soon if we don't consolidate this into one place.

Reply:
@hausdorff I've cleaned up the code a little, please take another look when you get the chance.